Porter.elm

```diff
@@ -1,4 +1,19 @@
-module Porter exposing (Model, Config, Msg, subscriptions, init, update, send)
+module Porter
+    exposing
+        ( Model
+        , Config
+        , Msg
+        , subscriptions
+        , init
+        , update
+        , send
+        , request
+        , andThen
+        , sendRequest
+        , map
+        , map2
+        , map3
```

Comment:
Should we export the `Request` type annotation in case someone wants to use it while defining chained requests defined top-level to a module?

Reply:
Can you describe this use case?

The reason I did not export the type annotation is so that people can _only_ define requests using `request` and at some point `sendRequest`.

Reply:
What I mean is a case when somebody wants to write something like this top-level to a module:
```elm
someRequest : ??
someRequest =
  Porter.request "abcd"
    |> Porter.andThen (\res -> Porter.request "efgh")
```
And then get frustrated about the compiler warning asking for a type annotation and not being able to give it one.
We can still hide the constructor by packing the `Request` further inside a single-constructor union type exporting only the type but not the constructor - thought there might be a better way here.

Reply:
So I'd want to do `exposing (Request)` (which currently does not occur because I missed it) but not `exposing Request(..)`.

Reply:
That should work, yes. To be 100% sure, would you mind verifying that you can't use it as a constructor?

Reply:
Verified! The Elm compiler will state 'Cannot find `Porter.Request`, did you mean `Porter.request`. Wonderful!